app/controllers/staff/transaction_uploads_controller.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require "csv"
+
+class Staff::TransactionUploadsController < Staff::BaseController
+  include Secured
+
+  before_action :authorize_report
+
+  def new
+    @report_presenter = ReportPresenter.new(@report)
+  end
+
+  def show
+    response.headers["Content-Type"] = "text/csv"
+    response.headers["Content-Disposition"] = "attachment; filename=transactions.csv"
+
+    csv = CSV.generate { |table|
```

Comment:
Did you consider using `CSV.generate_row` so you can write the CSV out to the stream one line at a time, rather than generating the whole thing and *then* streaming it out (which removes the streaming benenfit) ?

Reply:
I was principally using the streaming API because this data is dynamically generated, whereas the Rails `send_file` interface only seems to like sending a file from disk, although I suppose there's nothing to stop me just using `render` normally to send a blob of text.

Reply:
Where is `generate_row` documented? `ri` doesn't seem to know anything about it.

Reply:
I misremembered - it's `generate_line` - was previously `generate_row` back in Ruby 1.8.x days!

Reply:
Cool, I've amended that, thank you :)